assets/app/scripts/directives/oscFileInput.js

```diff
@@ -8,15 +8,83 @@ angular.module('openshiftConsole')
         model: "=",
         required: "=",
         disabled: "=ngDisabled",
-        helpText: "@?"
+        helpText: "@?",
+        dropArea: "@?"
       },
       templateUrl: 'views/directives/osc-file-input.html',
       link: function(scope, element){
         scope.helpID = _.uniqueId('help-');
         scope.supportsFileUpload = (window.File && window.FileReader && window.FileList && window.Blob);
         scope.uploadError = false;
+
+        var dropArea = scope.dropArea || element,
+        highlightDropArea = false,
+        showDropArea = false;
+
+        scope.$watch('disabled', function() {
+          if (scope.disabled) {
+            // Remove listeners for dropArea drag & drop actions
+            $(dropArea).off('dragover');
+            $(dropArea).off('dragleave');
+            $(dropArea).off('drop');
+          } else {
+            // Add listeners for dropArea drag & drop actions
+            $(dropArea).on('dragover', function() {
+              $(dropArea).addClass('highlight-drag-and-drop-area');
+              highlightDropArea = true;
+            });
+            $(dropArea).on('dragleave', function() {
+              highlightDropArea = false;
+              _.delay(function(){
+                  if( !highlightDropArea ){ $(dropArea).removeClass('highlight-drag-and-drop-area'); }
+              }, 200 );
```

Comment:
Does _.debounce or _.throttle not work? Then you don't need the `highlightDropArea` var.